tests/filtered_relation/tests.py

```diff
@@ -393,17 +393,19 @@ def test_difference(self):
         ).filter(book_jane__isnull=False)
         self.assertSequenceEqual(qs1.difference(qs2), [self.author1])
 
+    @skipUnlessDBFeature("has_select_for_update")
     def test_select_for_update(self):
-        self.assertSequenceEqual(
-            Author.objects.annotate(
-                book_jane=FilteredRelation(
-                    "book", condition=Q(book__title__iexact="the book by jane a")
-                ),
+        with transaction.atomic():
```

Comment:
In this nested atomic block really required?

Reply:
Without `transaction.atomic()` is crashes on MySQL with MyISAM :thinking: 
```
  File "/django/django/db/models/sql/compiler.py", line 657, in as_sql
    raise TransactionManagementError(
django.db.transaction.TransactionManagementError: select_for_update cannot be used outside of a transaction.
```

Reply:
Something is fishy here.

Ah I see what's happening. Since MyISAM doesn't support transactions (it basically ignores any `BEGIN` and `ROLLBACK` statement and is always auto-commit) then `TestCase` doesn't perform any form of transaction wrapping and thus the `select_for_update` call is not wrapped in an implicit `atomic` block like it normally is.

I see a few options:
1. Leave things as they are
2. Require that this test `supports_transactions`
3. Adjust the compiler to not  raise this error when `not connection.features.supports_transactions`

Ideally we should raise an error when `select_for_update` is attempted to be used on a backend that doesn't support transactions (as it's a noop) but that would be backward incompatible.

Reply:
I like the 3rd option, there are probably more related tests. Will move it to a separate PR.